crawler/main.py

```diff
@@ -44,76 +29,32 @@ def run(sftp: bool, keep_files: bool, add_to_dart: bool, settings_module: str =
-                # Index on unique combination of columns
-                logger.debug(f"Creating compound index on '{samples_collection.full_name}'")
-                # create compound index on 'Root Sample ID', 'RNA ID', 'Result', 'Lab ID' - some
-                # data had the same plate tested at another time so ignore the data if it is exactly
-                # the same
-                samples_collection.create_index(
-                    [
-                        (FIELD_ROOT_SAMPLE_ID, pymongo.ASCENDING),
-                        (FIELD_RNA_ID, pymongo.ASCENDING),
-                        (FIELD_RESULT, pymongo.ASCENDING),
-                        (FIELD_MONGO_LAB_ID, pymongo.ASCENDING),
-                    ],
-                    unique=True,
-                )
-
-                # Index on lh_source_plate_uuid column
-                # Added to make lighthouse API source completion event call query more efficient
-                logger.debug(f"Creating index '{FIELD_LH_SOURCE_PLATE_UUID}' on '{samples_collection.full_name}'")
-                samples_collection.create_index(FIELD_LH_SOURCE_PLATE_UUID)
-
-                if centre_prefix:
-                    # We are only interested in processing a single centre
-                    centres = list(filter(lambda config: config.get(CENTRE_KEY_PREFIX) == centre_prefix, centres))
-                else:
-                    # We should only include centres that are to be batch processed
-                    centres = list(
-                        filter(lambda config: config.get(CENTRE_KEY_INCLUDE_IN_SCHEDULED_RUNS, True), centres)
-                    )
-
-                centres_instances = [Centre(config, centre_config) for centre_config in centres]
-
-                for centre_instance in centres_instances:
-                    logger.info("*" * 80)
-                    logger.info(f"Processing {centre_instance.centre_config[CENTRE_KEY_NAME]}")
-
-                    try:
-                        if sftp:
-                            centre_instance.download_csv_files()
-
-                        centre_instance.process_files(add_to_dart)
-                    except Exception as e:
-                        logger.error(f"Error in centre '{centre_instance.centre_config[CENTRE_KEY_NAME]}'")
-                        logger.exception(e)
-                    finally:
-                        if not keep_files and centre_instance.is_download_dir_walkable:
-                            centre_instance.clean_up()
+            ensure_mongo_collections_indexed(db)
+
+            if centre_prefix:
+                # We are only interested in processing a single centre
+                centres = list(filter(lambda config: config.get(CENTRE_KEY_PREFIX) == centre_prefix, centres))
+            else:
+                # We should only include centres that are to be batch processed
+                centres = list(filter(lambda config: config.get(CENTRE_KEY_INCLUDE_IN_SCHEDULED_RUNS, True), centres))
```

Comment:
Do we have this flag setting in another pull request? (in config/defaults)?

Reply:
This is just old line 97 unindented as I removed it from a block that has been removed.  But yes, this was to support sending UAT test data in.  The centre that handles UAT test data creation shouldn't be looking for files on the SFTP to process on scheduled runs.

Reply:
Yes all centres have this key set.  It was something I added in summer last year alongside Philip.